app/js/enhancedDate.js

```diff
@@ -1,5 +1,239 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+/*
+ * Homework 
+ * JSCRIPT300-Spring2015/Module_2
+ * by Diane Zevenbergen
+ *
+ * Note: I assumed that the setDate method was supposed to be flexible and store 'theDate' either
+ * as a Date object OR in epoch milliseconds, depending on the argument that was passed.
+ * So wherever I use a Date object method such as getTime(), I call a fcn getDateObj to
+ * convert 'theDate' if needed.
+ */
 
+var enhancedDate = (function iife() {
+    
+    var theDate;
+    
+    /*
+     * This function checks whether 'theDate' has been set by the setDate
+     * method yet.  If not, it calls setDate & sets date to now.
+     * Called  by all methods in publicAPI (except setDate).
+     */
+    function verifyDate(){
+        if (typeof theDate === 'undefined') {
+            publicAPI.setDate();
+        }        
+    }
+    
+    // Function returns date object regardless if argument passed is epoch ms or Date obj
+    function getDateObj(date) {
+        if (date instanceof Date) {
+            return date;
+        } else {
+            return new Date(date);
+        }        
+    }
+
+    // Function returns epoch milliseconds regardless if argument passed is epoch ms or Date obj
+    function getDateMilli(date) {
+        if (date instanceof Date) {
+            return date.getTime();
+        } else {
+            return date;
+        }        
+    }
+    
+    var publicAPI = {
+        
+        // Store date passed in as Date object or epoch milliseconds (default to now)
+        setDate: function setDate(date) {
+            
+            // If no argument passed set date to now 
+            if (typeof date === 'undefined') {
+                date = Date.now();
+                
+            //  If date is NOT a Date obj OR a number (ms), set date to now
+            } else if (!(date instanceof Date) && (typeof date !== 'number')) {
+                date = Date.now();
+            }
+            
+            theDate = date;
+        },
+        
+        // Return date as Date object if argument is true, otherwise return milliseconds
+        getDate: function getDate(isDateObj) {
+            verifyDate();            
+
+            if (isDateObj) {
+                return getDateObj(theDate);
+            } else {
+                return getDateMilli(theDate);
+            }            
+        },
+        
+        // Return day of week ('Monday', etc.)
+        getDayName: function getDayName() {
+            var dayValue;
+            var dayLabel;
+
+            verifyDate();
+            
+            // Use getDay method on Date obj (getDateObj fcn converts from ms to obj if needed)
+            dayValue = getDateObj(theDate).getDay();
+            
+            // Assign weekday label based on weekday value
+            switch (dayValue) {
```

Comment:
Functions works as written, so my only feedback is just an implementation optimization. Notice how the getDay and getMonth return a number starting with 0? Think of how you can represent the day names and month names as values starting with 0 (hint: there's a particular data structure this maps very well to.) This would allow you to externalize configuration data like this as part of the private data of your module and the methods would have access to that data when returning the specific day or month string respectively.